node/consensus/src/lib.rs

```diff
@@ -278,11 +285,15 @@ impl<N: Network> Consensus<N> {
 impl<N: Network> Consensus<N> {
     /// Adds the given unconfirmed solution to the memory pool.
     pub async fn add_unconfirmed_solution(&self, solution: Solution<N>) -> Result<()> {
+        // Calculate the transmission checksum.
+        let checksum = Data::<Solution<N>>::Buffer(solution.to_bytes_le()?.into()).to_checksum::<N>()?;
```

Comment:
Slight efficiency improvement, the checksum calculation can be propagated one level higher. This function is called from two places, in one of those `fn unconfirmed_solution`, we can call `serialized.solution.to_checksum()`